### PR TITLE
Add PWA requirements

### LINKED ISSUE #278: Add PWA Support
Add [Vite PWA](https://vite-pwa-org.netlify.app/) support to the rewrite

### PR DESCRIPTION
**Pull request by bastislack**
Make the app an installable PWA.
Closes #278.

### CODE REVIEW

**Comment by WaldemarLehner** [comment]
Lighthouse PWA Report: 
image

Only thing thats marked as an issue here is that the [Manifest doesn't have a maskable icon](https://developer.chrome.com/docs/lighthouse/pwa/maskable-icon-audit/?utm_source=lighthouse&utm_medium=devtools)

Given that we don't have an icon configured yet, I wouldn't bother with this for now.

File: public/logo96.png
Comment (WaldemarLehner) [inline public/logo96.png:1]: Icon is not suitable for small screens. But that's an issue for later ;)

**Review by WaldemarLehner — APPROVED** [review]

File: pnpm-lock.yaml
Comment (WaldemarLehner) [inline pnpm-lock.yaml:1]: Actually… this shouldnt be here. We make use of package-lock.json — having 2 "sources of truth" will just make things break i.e. "it works on my machine" — we either use pnpm, or npm.

**Review by WaldemarLehner — CHANGES_REQUESTED** [review]

File: pnpm-lock.yaml
Comment (bastislack) [reply pnpm-lock.yaml:1]: I read that pnpm is better than npm. Should we switch to using pnpm, or should we just stick to npm?

Comment (WaldemarLehner) [reply pnpm-lock.yaml:1]: It's "better" in the sense that it makes use of centralized cache locally and just sym-links to that cache. 

**BUT** it also means that anyone that wants to work on the Repo *has* to use pnpm — which imo is not something we want.

I personally would like to stick with the default node package manager, npm :D

**Pull request merged**